File: src/main/java/com/comma/newsbot/domain/News.java
package com.comma.newsbot.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "news")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(nullable = false, length = 850, unique = true)
    private String url;

    @Column(nullable = false)
    private String description;

    private LocalDateTime publishedAt;

    private LocalDateTime processedAt;
}

File: src/domain/news.py
from sqlalchemy import Column, BigInteger, String, DateTime
from src.database import Base
from datetime import datetime


class News(Base):
    __tablename__ = "news"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    title = Column(String(500), nullable=False)
    url = Column(String(850), nullable=False, unique=True)
    description = Column(String, nullable=False)
    published_at = Column(DateTime, nullable=True)
    processed_at = Column(DateTime, nullable=True, default=datetime.now)
